include/date/tz.h

```diff
@@ -43,20 +43,18 @@
 // required. On Windows, the names are never "Standard" so mapping is always required.
 // Technically any OS may use the mapping process but currently only Windows does use it.
 
+#include "tz_config.h"
+
 #ifndef USE_OS_TZDB
-#  define USE_OS_TZDB 0
+#  error "Missing define expected from configuration header"
 #endif
```

Comment:
Sorry, can't do these changes to tz.h.  CMake is just _a_ build system.  It is not _the_ build system.

I want to keep it easy to configure this lib from the command line by using `-DMACRO=VALUE`, and to have reasonable default values for the macros.

Reply:
I see, but are you still okay with `#include "tz_config.h"` then?

Reply:
The thing I'm concerned about is, when I use this in a larger project, if we build `tz.cpp` with one set of macros, and include it in a larger project using a different set of macros, we just get undefined behavior, and this caused a bunch of crashes in our firmware.

When cmake chooses values for those macros and then compiles tz.cpp and installs it, I want to still have a source of truth for what those values are, otherwise, it's a bit risky to link into the project

Are you okay with an `if has_include "tz_config.hpp"` or whatever the right syntax for that is? I think it might be possible on gcc and clang, I don't know if it's standard, I can't remember

Reply:
`__has_include` was standardized in C++17.  This project still has to support C++11 and C++14.  If there's a way to do that for clang, gcc and VC++, I'm fine with that.  I haven't investigated whether there is such a way or not.

I'm ok with adding a config file.  But I want to keep the ability to build from the command line for projects:

*  That just use date.h.
*  That use tz.h.
*  That just use iso_week.h, islamic.h, julian.h, or ptz.h.

CMake support has to be a pure add-on to this lib.  Also, the less files the better.  In many places I've already advertised that if you just need civil calendar support, date.h is all you need.  It would be nice to keep that.

Reply:
Can "build from command line" procedure include "copy `config.h.in` from the root of the library and adjust as needed for your project, and put it in the include path"?

Or you want a different tool that does that?

I guess we could do something like pass one define `-DTZ_USE_CONFIG_HEADER` that causes it to include `tz_config.h`, and otherwise it doesn't do it. That would still be safer than trying to get all the configuration options right in two different places. Does that sound good to you?